src/abstract-interpretation/data-frame/absint-visitor.ts

```diff
@@ -142,12 +142,12 @@ export class DataFrameShapeInferenceVisitor<
 		const value = resolveIdToDataFrameShape(node.info.dataFrame.expression, this.config.dfg, this.newDomain);
 
 		if(value !== undefined) {
-			this.newDomain.set(node.info.dataFrame.identifier, value);
+			this.newDomain.value.set(node.info.dataFrame.identifier, value);
```

Comment:
why do we allow a direct access to the underlying value map? similar to `create`, `join` etc. i would prefer this to be wrapped in an `assign` or something similar